src/units/abilities.cpp

```diff
@@ -228,12 +220,9 @@ bool unit::get_ability_bool(const std::string& tag_name, const map_location& loc
 	// different from the central unit, that the ability is of the right type, detailed verification of each ability),
 	// if so return true.
 	for(const unit& u : units) {
-		if(!u.has_ability_distant() || u.incapacitated() || same_unit(u, *this) || !u.affect_distant(tag_name)) {
```

Comment:
Some of these checks will be faster than others, some will return false more often than others, and although they're all going to be fast, I assume this code path is going to be heavily used during the AI turn. Assuming most units won't have `[affects_adjacent]` abilities, I feel it's good to check `u.has_ability_distant()` before even calculating `distance_between`.